### ISSUE TITLE
Rename doesn't resolve conflicts by inserting identity conversion

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#6308]
```C#
[Obsolete]
void M<T>(T t, Func<T,int, int> e) { }
int M<T>(T t, Func<T, long, long> e) => M(new { }, (_, a) => X(a));

long X(long a) => a;
int Y(int a) => a;
```

Rename `Y` to `X`.

**Expected**:
```C#
M(new { }, (_, a) => X((long)a));
```

**Comment by Pilchie** [comment]
This is actually an issue in remove redundant cast, which flows through into rename.

**Comment by SLaks** [comment]
IIRC, the cast in the expected case is not reported as redundant (I don't have VS at the moment)

**Comment by DustinCampbell** [comment]
It is:

![image](https://cloud.githubusercontent.com/assets/116161/10765722/60efd6ca-7c91-11e5-9075-85427947989c.png)

**Comment by DustinCampbell** [comment]
#6432 fixes this issue as well. The problem is that the expander was incorrectly attempting to add types to lambda parameters and generic type arguments that were actually anonymous types. This caused the simplifier to work in unexpected ways. The conflict resolution is slightly different but still effective:

![image](https://cloud.githubusercontent.com/assets/116161/10955318/e09a3a38-8307-11e5-8e23-700e6e0de612.png)
